### PR TITLE
Update serialize/deserialize to use AS builtins

### LINKED ISSUE #24: Migrate to using fromUTF8 and toUTF-8 provided by assemblyscript
Our friends at AS have been busy and have implemented this new functionality for strings. We should replace the sketchy serialize/deserialize code that assumes only single byte UTF-8 is being passed

### PR DESCRIPTION
**Pull request by willemolding**
Closes #24

### CODE REVIEW

File: utils.ts [hunk 1]
```diff
@@ -42,33 +42,17 @@ export function check_encoded_allocation(encoded_allocation: u32): ErrorCode {
...
-  // each char takes two bytes, encoded
-  let ptr = memory.allocate(dataLength << 1);
-  //checkMem();
-  for (let i = 0; i < dataLength; ++i) {
-    store<u16>(ptr + i, val.charCodeAt(i));
-  }
-  let encoded_allocation = u32_merge_bits(ptr as u16, dataLength as u16);
-  return encoded_allocation;
+  let ptr = val.toUTF8();
+  let length = val.length;
```
Comment (MaxGraey) [inline utils.ts]: Are you need length in UTF16 or UTF8? In last case you should use `val.lengthUTF8` getter instead `val.length`

Comment (willemolding) [reply utils.ts]: Good save. This would have been disastrous for multi-byte characters

Comment (Connoropolous) [reply utils.ts]: :pray

Comment (Connoropolous) [reply utils.ts]: actually, just based on testing, it appears that the way you have it works @willemolding, and the way it's suggested doesn't

**Comment by willemolding** [comment]
The way it was originally will fail on multi-byte characters since the length will not be equal to the number of characters and the string will not decode correctly. 

I have found some inconsistencies in how holochain core is passing string (see comments). To satisfy this I am checking if the last byte is null. This should be ok for now until we sort out the problem.

Also, I'm having troubles with getting the tests running at the moment so best check that this works before merging.

**Comment by willemolding** [comment]
I also suggest we add some test that attempt to pass and receive multi-byte chars from holochain to ensure we have that covered

**Comment by Connoropolous** [comment]
@willemolding what's an example of a multi-byte character? Can you add tests to cover those cases?

**Comment by willemolding** [comment]
Ok I added some tests cases and I'm happy with it. I'll go ahead and merge

**Pull request merged**